includes/multi-currency/Notes/NoteMultiCurrencyAvailable.php

```diff
@@ -51,4 +59,30 @@ public static function get_note() {
 
 		return $note;
 	}
+
+	/**
+	 * Add the note if it passes predefined conditions.
+	 */
+	public static function possibly_add_note() {
+		// Don't add the note if the merchant didn't create a WCPay account yet.
+		if ( is_null( self::$account ) || ! self::$account->is_stripe_connected() ) {
+			return;
+		}
```

Comment:
However, I see that you've removed relevant tests too. I think it's worth adding tests again to cover the logic here. 

We can use `NoteTraits::note_exists()` to verify if the note has been added or not. WDYT?

Reply:
Sounds good, I added back some tests to cover this piece of logic, thanks!